src/main/java/org/minions/devfund/katerinaanzoleaga/moviedatabase/MovieDatabase.java

```diff
@@ -0,0 +1,280 @@
+        if (this.movieExists(name) == -1) {
+            Movie aMovie = new Movie(name);
+            this.movieList.add(aMovie);
+            for (String actor : actors) {
+                int i = this.actorExists(actor);
+                if (i == -1) {
+                    Actor anActor = new Actor();
+                    anActor.setName(actor);
+                    aMovie.addActor(anActor);
+                    anActor.addMovie(aMovie);
+                    this.actorList.add(anActor);
+                }
+                else{
+                    aMovie.addActor(this.actorList.get(i));
+                    this.actorList.get(i).addMovie(aMovie);
+                }
+            }
+        }
+    }
+
+    /**
+     * Adds a rating to a movie with the name specified only if the rating is zero.
+     * @param name
+     * @param rating
+     */
+    void addRating(String name, double rating) {
+        int i = this.movieExists(name);
+        if (i != -1) {
+            Movie aMovie;
+            aMovie = this.movieList.get(i);
+            if (aMovie.getRating() == 0) {
+                aMovie.setRating(rating);
+            }
+        }
+
+    }
+
+    /**
+     * Updates the rating of a movie regardless of its original value.
+     * @param name
+     * @param newRating
+     */
+    void updateRating(String name, double newRating) {
+        // The following gets the movie index and updates the rating
+        int i = this.movieExists(name);
+        if (i != -1) {
+            Movie aMovie;
+            aMovie = this.movieList.get(i);
+            aMovie.setRating(newRating);
+        }
+    }
+
+    /**
+     * Gets the Actor with best average rating.
+     * @return
+     */
+    String getBestActor() {
+        double maxAvrRating = 0;
+        double avrRating = 0;
+        int index = -1;
+        Actor theActor;
+        for (Actor actor: this.actorList) {
+            avrRating = actor.getAverageRating();
+            if (avrRating > maxAvrRating) {
+                maxAvrRating = avrRating;
+                index = this.actorList.indexOf(actor);
+            }
+        }
+        theActor = this.actorList.get(index);
+        return theActor.getName();
+    }
+
+    /**
+     * Gets the best movie based on the max rating.
+     * @return
+     */
+    String getBestMovie() {
+        double maxRating = 0;
+        int index = -1;
+        Movie theMovie;
+        for (Movie movie : this.movieList) {
+            if (movie.getRating() > maxRating){
+                maxRating = movie.getRating();
+                index = this.movieList.indexOf(movie);
+            }
+        }
+        theMovie = this.movieList.get(index);
+        return theMovie.getName();
+
+    }
+
+
+    /**
+     * Fill Movies list from file.
+     * @param fileName
+     */
+    void fillMoviesFromFile(String fileName) {
+        try {
+            File moviesFile = new File(fileName);
+            Scanner scnr = new Scanner(moviesFile);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Scanner" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)